### PR TITLE
landing-page-pagination

### PR DESCRIPTION
**Pull request by Serhio46**

### CODE REVIEW

File: workflow/landing-page/js/loader.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//Кнопки пагинации
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:1]: https://www.youtube.com/watch?v=bUa7xJWuz_s

File: workflow/landing-page/js/loader.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		currentOffset -= itemsPerPage;
+	}
+	fetchHandler()
+});
+
+//Запрос на сервре и изменение контена
+
+let currentOffset = 0;
+let itemsPerPage = 9;
+let totalCount = 4000;//Хотел сюда взять по общему числу картинок(стр.17), но видно ограничения на бесплатной версии, поставил 4000, что бы работало при листании назад сразу!
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:23]: use const if no reassigning

File: workflow/landing-page/js/loader.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+async function fetchHandler() {
+	try {
+		let gifs = await fetch(`http://api.giphy.com/v1/gifs/search?q=cats&api_key=6yeW4Y3DQYvcNrhsWv8h741VsbL9kwUV&limit=${itemsPerPage}&offset=${currentOffset}`)
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:27]: use const if no reassigning

File: workflow/landing-page/js/loader.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			.then((resp) => resp.json())
+			.then((data => {
+				if (totalCount === 0) {
+					//totalCount = data.pagination.total_count; // Вот тут добавлял
+				}
+				return data.data
+			}));
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:34]: use only one approach - callbacks or async/await

File: workflow/landing-page/js/loader.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+let totalCount = 4000;//Хотел сюда взять по общему числу картинок(стр.17), но видно ограничения на бесплатной версии, поставил 4000, что бы работало при листании назад сразу!
+
+async function fetchHandler() {
+	try {
+		let gifs = await fetch(`http://api.giphy.com/v1/gifs/search?q=cats&api_key=6yeW4Y3DQYvcNrhsWv8h741VsbL9kwUV&limit=${itemsPerPage}&offset=${currentOffset}`)
+			.then((resp) => resp.json())
+			.then((data => {
+				if (totalCount === 0) {
+					//totalCount = data.pagination.total_count; // Вот тут добавлял
+				}
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:32]: useless if

File: workflow/landing-page/js/loader.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+				return data.data
+			}));
+		changeCart(gifs);
+	}
+	catch (error) {
+		console.log('Ошибка загрузки данных')
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:38]: no russian

File: workflow/landing-page/js/loader.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+//Функция изменения контента
+function changeCart(gifs) {
+	let carts = document.querySelectorAll('.item-story');
+
+	let date = new Date();
+	let min = date.getMinutes();
+	let hours = date.getHours();
+	let month = date.getMonth() + 1;
+	let dat = date.getDate();
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:51]: use `const` if no reassigning

Comment (qwaseras) [inline workflow/landing-page/js/loader.js:51]: use informative names for variables

File: workflow/landing-page/js/loader.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+	const change = (current) => {
+		if (current <= 9) {
+			return `0${current}`
+		} return current
```
Comment (qwaseras) [inline workflow/landing-page/js/loader.js:56]: move `return` to the next line

**Review by qwaseras — CHANGES_REQUESTED** [review]

File: workflow/landing-page/js/loader.js (hunk 1 above)
Comment (Serhio46) [reply workflow/landing-page/js/loader.js:1]: I've played this game)

**Pull request merged**